core/src/main/java/org/jdbi/v3/core/argument/ArrayElementMapper.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core.argument;
+
+/**
+ * Maps array elements into SQL array elements compatible with a specific JDBC driver.
+ * @param <T> the element type being mapped/converted.
+ */
+public interface ArrayElementMapper<T> {
```

Comment:
It's not really a builder though, it's a strategy used by `SqlArrayArgumentFactory` to create the `Object[]` that'll be converted to a SQL array.